packages/core/src/lib/base_protocol.ts

```diff
@@ -50,6 +56,52 @@ export class BaseProtocol implements IBaseProtocol {
     );
     return peer;
   }
+
+  protected async getPeers(peerId?: PeerId): Promise<Peer[]> {
+    const selectedPeers: Peer[] = [];
+
+    const { peer } = await selectPeerForProtocol(
+      this.peerStore,
+      [this.multicodec],
+      peerId,
+    );
+
+    selectedPeers.push(peer);
+
+    // only return multiple peers for protocols LP and Filter
+    if (
+      ![Codecs.LightPush, Codecs.FilterPush, Codecs.FilterSubscribe].includes(
```

Comment:
You are mixing up inheriter and inheritee logic here by assuming that light push, etc would inherit this class and that a specific logic applies to them.

Instead, have this `getPeers` accept an argument that set the number of peer you want returned. You can then bind this function in the inheriter (e.g. `Filter`) to a given value.

Reply:
Also, forgot about relay. For relay, you probably want to connect to all the peers you discover.